sdk/python/kubeflow/storage_init_container/storage.py

```diff
@@ -27,16 +27,26 @@ def dataset_factory(dataset_provider, dataset_provider_args):
     parser = argparse.ArgumentParser(
         description="script for downloading model and datasets to PVC."
     )
-    parser.add_argument("model_provider", type=str, help="name of model provider")
     parser.add_argument(
-        "model_provider_args", type=str, help="model provider serialised arguments"
+        "--model_provider", type=str, help="name of model provider", required=False
+    )
+    parser.add_argument(
+        "--model_provider_parameters",
+        type=str,
+        help="model provider serialised arguments",
+        required=False,
```

Comment:
The default value for required is False, so I think we can remove required=False from the `add_argument`